pokemon-league-participation-manager/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/dao/PokemonDAOTest.java

```diff
@@ -0,0 +1,168 @@
+        pokemon1.setTrainer(trainer);
+        pokemonDao.createPokemon(pokemon1);
+
+        pokemon2 = new Pokemon();
+        pokemon2.setNickname("Pokemon 2");
+        pokemon2.setDateTimeOfCapture(LocalDateTime.of(2018, Month.MARCH, 1, 0, 1));
+        pokemon2.setSpecies(species);
+        pokemon2.setTrainer(trainer);
+        pokemonDao.createPokemon(pokemon2);
+
+    }
+
+    /**
+     * Test of createPokemon method with valid imput.
+     */
+    @Test
+    public void testCreateValidPokemon() {
+        Pokemon newPokemon = new Pokemon();
+        newPokemon.setNickname("Pokemon 2");
+        newPokemon.setDateTimeOfCapture(LocalDateTime.of(2018, Month.MARCH, 1, 0, 2));
+        newPokemon.setSpecies(species);
+        newPokemon.setTrainer(trainer);
+        pokemonDao.createPokemon(newPokemon);
+
+        List<Pokemon> pokemons = pokemonDao.getAllPokemon();
+        Assert.assertEquals(3, pokemons.size());
+        Assert.assertTrue(pokemons.contains(newPokemon));
+    }
+
+    /**
+     * Test of createPokemon method with missing mandatory name.
+     */
+    @Test
+    public void testCreatePokemonWithoutName() {
+        Pokemon newPokemon = new Pokemon();
+
+        assertThatExceptionOfType(PersistenceException.class).isThrownBy(() -> pokemonDao.createPokemon(newPokemon));
+
+    }
+
+    /**
+     * Test of createPokemon method with null as imput.
+     */
+    @Test
+    public void testCreateNullPokemon() {
+
+        Pokemon newPokemon = new Pokemon();
+
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> pokemonDao.createPokemon(null));
+
+    }
+
+    /**
+     * Test of updatePokemon method
+     */
+    @Test
+    public void testUpdatePokemon() {
+        Pokemon pokemonToUpdate = pokemonDao.findPokemonById(pokemon1.getId());
+        pokemonToUpdate.setNickname("Changed NickName");
+
+        pokemonDao.updatePokemon(pokemonToUpdate);
+
+        Pokemon found = pokemonDao.findPokemonById(pokemon1.getId());
+        Assert.assertEquals(found, pokemonToUpdate);
+        Assert.assertEquals("Changed NickName", found.getNickname());
+    }
+
+    /**
+     * Test of deletePokemon method
+     */
+    @Test
+    public void testDeletePokemon() {
+        pokemonDao.deletePokemon(pokemon1);
+
+        List<Pokemon> pokemons = pokemonDao.getAllPokemon();
+        Assert.assertEquals(1, pokemons.size());
+    }
+
+    /**
+     * Test of findPokemonById method
+     */
+    @Test
+    public void testFindPokemonById() {
+        Pokemon found = pokemonDao.findPokemonById(pokemon1.getId());
+        Assert.assertEquals(pokemon1, found);
+    }
+
+    /**
+     * Test of getAllPokemon method
+     */
+    @Test
+    public void testGetAllPokemon() {
+        List<Pokemon> pokemons = pokemonDao.getAllPokemon();
+
+        Assert.assertEquals(2, pokemons.size());
+        Assert.assertTrue(pokemons.contains(pokemon1));
+        Assert.assertTrue(pokemons.contains(pokemon2));

```

Comment:
This can be simplified in this way assertThat(pokemonDao.getAllPokemon()).usingFieldByFieldElementComparator().containsOnly(pokemon1, pokemon2);